Sources/ProjectAutomation/Settings.swift

```diff
@@ -0,0 +1,18 @@
+import Foundation
+
+// A group of settings configurations.
+
+public typealias BuildConfigurationDictionary = [ProjectAutomation.BuildConfiguration: ProjectAutomation.Configuration]
+
+public struct Settings: Equatable, Codable {
```

Comment:
This is very different from the `Settings` we expose in `ProjectDescription` or `TuistGraph`.
Even if we are not interested in some of the pieces, I would make the format compatible (i.e. the ones that we need should have the same information), or we won't be able to expand on it without breaking the API

Reply:
will adapt to same structure

Reply:
done

Reply:
Here settings is still an array of configurations, while it's a completely different type in ProjectDescription/TuistGraph, isn't it?

Reply:
it is a dictionary, then mapped to a array of configurations

Reply:
I mean what you get out is an array with just the keys of the dictionary, what if in the future we need to add also the values? it would be a breaking change, isn't it?
Wouldn't be better to provide the dictionary already?

Reply:
ProjectDescription.Settings also has:
let configurations: [Configuration]

Reply:
what is the benefit in
["debug": "debug", "release": "release"]
of ["release", "debug"] ?

Reply:
as mentioned the external graph is not expected to be stable, if needed we can break

Reply:
The value would contain the settings (now or in a future expansion), not just the name.

> the external graph is not expected to be stable, if needed we can break

That's not true I think. If we expose it through PackageAutomation, then we can't break it or we would break plugins.
What we can break freely is TuistGraph

Reply:
changed to new structure